### DISCUSSION TITLE
v3.1.0

### DISCUSSION THREAD

**Discussion by njzjz** [deepmodeling/deepmd-kit#4799] [Announcement]
### DPA3

DPA3 is an advanced interatomic potential leveraging the message-passing architecture. Designed as a large atomic model (LAM), DPA3 is tailored to integrate and simultaneously train on datasets from various disciplines, encompassing diverse chemical and materials systems across different research domains. Its model design ensures exceptional fitting accuracy and robust generalization within and beyond the training domain. Furthermore, DPA3 maintains energy conservation and respects the physical symmetries of the potential energy surface, making it a dependable tool for a wide range of scientific applications.

Refer to `examples/water/dpa3/input_torch.json` for the training script. After training, the PyTorch model can be converted to the JAX model.

### PaddlePaddle backend

The PaddlePaddle backend features a similar Python interface to the PyTorch backend, ensuring compatibility and flexibility in model development. PaddlePaddle has introduced dynamic-to-static functionality and PaddlePaddle JIT compiler (CINN) in DeePMD-kit, which allow for dynamic shapes and higher-order differentiation. The dynamic-to-static functionality automatically captures the user’s dynamic graph code and converts it into a static graph. After conversion, the CINN compiler is used to optimize the computational graph, thereby enhancing the efficiency of model training and inference. In experiments with the DPA-2 model, we achieved approximately a 40% reduction in training time compared to the dynamic graph, effectively improving the model training efficiency.

### Breaking changes
* breaking: enable PyTorch backend for PyPI LAMMPS by @njzjz in https://github.com/deepmodeling/deepmd-kit/pull/4728

### Other new features
* feat(pt/dp): support case embedding and sharable fitting by @iProzd in https://github.com/deepmodeling/deepmd-kit/pull/4417
* feat(pt): train with energy Hessian by @1azyking in https://github.com/deepmodeling/deepmd-kit/pull/4169
* feat: add new batch size rules for large systems by @caic99 in https://github.com/deepmodeling/deepmd-kit/pull/4659
* feat: add method to access fele in pppm/dplr by @HanswithCMY in https://github.com/deepmodeling/deepmd-kit/pull/4452
* feat (tf/pt): add atomic weights to tensor loss by @ChiahsinChu in https://github.com/deepmodeling/deepmd-kit/pull/4466
* feat(pt): add `trainable` to property fitting by @ChiahsinChu in https://github.com/deepmodeling/deepmd-kit/pull/4599 
* Feat(pt): Support fitting_net input statistics. by @Chengqian-Zhang in https://github.com/deepmodeling/deepmd-kit/pull/4504
* feat(jax): Hessian by @njzjz in https://github.com/deepmodeling/deepmd-kit/pull/4649
* feat: add plugin mode for data modifier by @ChiahsinChu in https://github.com/deepmodeling/deepmd-kit/pull/4621
* feat(pt): add eta message for pt backend by @HydrogenSulfate in https://github.com/deepmodeling/deepmd-kit/pull/4725
* feat: add huber loss by @iProzd in https://github.com/deepmodeling/deepmd-kit/pull/4684
* feat(pt): add AdamW for pt training by @iProzd in https://github.com/deepmodeling/deepmd-kit/pull/4757
* Feat:support customized rglob by @anyangml  in https://github.com/deepmodeling/deepmd-kit/pull/4763
* feat(pt/pd): add size option to dp show by @iProzd in https://github.com/deepmodeling/deepmd-kit/pull/4783

All changes in v3.0.1, v3.0.2, and v3.0.3 are included.

## Contributors

* @iProzd #4417 #4655 #4419 #4609 #4633 #4647 #4675 #4684 #4730 #4757 #4754 #4756 #4760 #4778 #4781 #4783 #4792
* @pre-commit-ci #4420 #4449 #4464 #4473 #4497 #4521 #4539 #4552 #4566 #4574 #4579 #4596 #4602 #4611 #4645 #4660 #4672 #4690 #4699 #4708 #4712 #4719 #4723 #4736 #4748 #4767 #4779 #4791
* @njzjz #4482 #4483 #4484 #4507 #4619 #4410 #4438 #4442 #4446 #4459 #4485 #4479 #4508 #4534 #4531 #4542 #4550 #4553 #4557 #4561 #4565 #4570 #4575 #4547 #4582 #4613 #4624 #4558 #4638 #4636 #4640 #4649 #4668 #4680 #4720 #4728 #4738 #4692 #4700 #4704 #4702 #4717 #4724 #4726 #4729 #4735 #4753 #4774 #4765 #4776 #4775 #4766 #4780 #4786 #4794
* @Chengqian-Zhang #4471 #4504 #4639
* @HydrogenSulfate #4418 #4489 #4673 #4302 #4439 #4414 #4480 #4493 #4488 #4512 #4467 #4514 #4617 #4556 #4656 #4694 #4701 #4715 #4725 #4768 #4770
* @QuantumMisaka #4510
* @1azyking #4169
* @caic99 #4535 #4615 #4659 #4434 #4426 #4435 #4433 #4437 #4463 #4505 #4478 #4541 #4513 #4597 #4622 #4662 #4669 #4677 #4678 #4688 #4687 #4737 #4747 #4746 #4761 #4772 #4773 #4784 #4751 #4790
* @dependabot #4408 #4630
* @anyangml #4423 #4432 #4587 #4763
* @HanswithCMY #4452
* @ChiahsinChu #4466 #4538 #4599 #4621
* @RMeli #4577
* @Yi-FanLi #4581
* @wanghan-iapcm #4653
* @SumGuo-88 #4593
* @SigureMo #4664
* @njzjz-bot #4796

## New Contributors
* @HanswithCMY made their first contribution in https://github.com/deepmodeling/deepmd-kit/pull/4452
* @QuantumMisaka made their first contribution in https://github.com/deepmodeling/deepmd-kit/pull/4510
* @1azyking made their first contribution in https://github.com/deepmodeling/deepmd-kit/pull/4169
* @RMeli made their first contribution in https://github.com/deepmodeling/deepmd-kit/pull/4577
* @SumGuo-88 made their first contribution in https://github.com/deepmodeling/deepmd-kit/pull/4593
* @SigureMo made their first contribution in https://github.com/deepmodeling/deepmd-kit/pull/4664

**Full Changelog**: https://github.com/deepmodeling/deepmd-kit/compare/v3.0.0...v3.1.0rc0

<hr /><em>This discussion was created from the release <a href='https://github.com/deepmodeling/deepmd-kit/releases/tag/v3.1.0'>v3.1.0</a>.</em>
